src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/ReflectionExtensions.cs

```diff
@@ -98,6 +93,11 @@ public static bool TryUnboxResult(this object? value, out object? result)
                     return true;
                 }
 
+                if (result is Task)
+                {
+                    throw new ArgumentException("Should never need to unbox a Task");
```

Comment:
Would provide a better message than this e.g.

```cs
throw new ArgumentException(
    "Cannot unbox Tasks as this may cause deadlock. Consider awaiting the result first.");
```

Reply:
Also changed the exception message in `TryBoxToResult`.